### DISCUSSION TITLE
Referencias para basarse

### DISCUSSION THREAD

**Discussion by luchopcerra** [wakeuplabs-io/wake3#1] [Ideas]
https://docs.semaphore.pse.dev/getting-started

**Comment by luchopcerra** [comment]
https://ui.shadcn.com/

**Comment by matzapata** [comment]
Once used this one https://infinitered.github.io/gluegun/#/toolbox-template I think aws and vue use it, cool the toolbox, already made utils for templating and installing deps and so on

**Comment by jeremylongshore** [comment]
Hey there! Great question - this definitely looks confusing at first glance.

The short answer: **No, they don't concatenate**. The paths work differently than you might expect.

## How It Actually Works

- **`pluginRoot`**: This is metadata *about* the marketplace structure. It tells tools "hey, plugins generally live in the `./plugins` directory." Think of it as documentation.

- **`source`**: This is the **absolute path from the repository root** to the specific plugin. It's not relative to `pluginRoot`.

So when Claude Code processes this:
```json
{
  "pluginRoot": "./plugins",
  "source": "./plugins/productivity/000-jeremy-content-consistency-validator"
}
```

It uses the `source` path directly: `./plugins/productivity/000-jeremy-content-consistency-validator`

It does **not** concatenate them into `./plugins/plugins/...`

## Why Have Both?

Good question! The `pluginRoot` is mostly for:
1. **Marketplace metadata** - Helps tools understand the repo structure
2. **Documentation** - Makes it clear where plugins live
3. **Future tooling** - Could be used for validation scripts, etc.

You're right that it looks redundant since every `source` path starts with `./plugins` anyway. This is a quirk of the schema design.

## For Your Own Marketplace

If you're building your own plugin marketplace, you can safely use:
- **`pluginRoot`**: Set it to where your plugins live (`./my-plugins`, etc.)
- **`source`**: Always use the full path from repo root

Hope this clears it up! Let me know if you have other questions - happy to help as you're building your own plugins.

— Jeremy
